### ISSUE TITLE
Questions about applying mix-resolution-vit to other downstream tasks such as semantic segmentation.

### ISSUE THREAD

**Issue by UCASHurui** [TomerRonen34/mixed-resolution-vit#2]
Hi, thanks for this elegant work. 

I am working on a weakly supervised semantic segmentation(WSSS) project and it will be interesting to substitute the default patch embed modules with the mix-resolution-vit. WSSS methods usually depend on the CAM generated by the model where the patch tokens have to be reshaped back to the shape of image features(in patch scale) if ViT-based models are employed. Since a token of mix-resolution-vit represents a patch of arbitrary size, it is unnatural to perform such flattening process. I am wondering if you could share some experiences to address this issue.

By the way, it would be appreciated if the fine-tuned checkpoints could be shared. Thanks in advance!

**Comment by TomerRonen34** [comment]
Hi,

Sorry about the delay regarding finetuned checkpoints, I'll try to release
them soon.

Regarding the usage of mixed-res ViTs for segmentation, I have 2 ideas, let
me know if they're helpful:

1. For each token embedding, predict a segmentaion mask that is the size of
the original patch – 16x16, 32x32 or 64x64 pixels.

2. Alternatively, perform a "reverse Quadtree" operation at the end of the
ViT, which creates a token volume of size [image_size/16, image_size/16,
embed_dim] where each token embedding is duplicated to completely fill its
corresponding space, e.g. tokens that represent a 64x64 patch will be
duplicated 16 times to create a 4x4 token grid.
I have code that does that in JAX, I can convert it to PyTorch and release
it if it's useful to you.

**Comment by UCASHurui** [comment]
Thanks for your reply! I will definitely try these two methods. I have tried an approach similar to the "reverse Quadtree”, which did not work well maybe due to my incorrect implementation. I am looking forward to your "reverse Quadtree” code.

**Comment by TomerRonen34** [comment]
Hi Rui,
I released an efficient torch implementation of the Reverse Quadtree
operation, including a sanity check that validates correctness. Check them
out at mixed_res/quadtree_impl/reverse_quadtree.py
and examples/04_reverse_quadtree.ipynb, and please let me know if it's
useful :)
